src/pathfinder/yapf/yapf_road.cpp

```diff
@@ -54,6 +55,41 @@ class CYapfCostRoadT
 		return 0;
 	}
 
+	struct TileDataHelper
+	{
+		TileIndex tile;    ///< The tile that yapf is calculating the cost for.
+		DiagDirection dir; ///< The direction the vehicle is travelling.
+	};
+
+	/* Find if the tile entry direction is blocked, preventing an RV from stopping. */
+	static Vehicle *FindBlockingVehiclesInTile(Vehicle *v, void *data)
+	{
+		TileDataHelper *tdh = (TileDataHelper*)data;
+
+		/* Exclude if not a RV, not travelling in the right direction or moving */
+		if (v->type != VEH_ROAD || DirToDiagDir(v->direction) != tdh->dir || !v->IsPrimaryVehicle() || v->cur_speed > 0) return nullptr;
```

Comment:
This appears to only detect road vehicles where the front part is on the entrance tile.
In the case where the road stop and the road vehicle are more than one tile long, this may not detect that the stop is blocked at all, even when it is.

Reply:
I see, yes a 3 carriage tram could be on the middle of the 2nd stop tile and still blocking the entrance. I had intended to focus on only vehicles that were loading in the destination, but I'll adjust it to consider all vehicles and then only the length of that carriage.